src/preproc/preprocessing_pass_api.h

```diff
@@ -0,0 +1,45 @@
+/*********************                                                        */
+/*! \file preprocessing_pass_api.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **  Justin Xu
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Preprocessing pass api for passes
+ **
+ ** Preprocessing pass api for passes. Takes in variables and methods from d_smt
+ ** and d_private and allows for access to the passes.
```

Comment:
d_smt and d_private are private variables, they should not be mentioned here. How about some description like:

"The preprocessing pass API is the interface between the rest of the solver and preprocessing passes. Passes are expected to use the API mostly for read-only access."